extensions/rich_text_components/Video/Video.js

```diff
@@ -25,16 +25,44 @@ oppia.directive('oppiaNoninteractiveVideo', [
       restrict: 'E',
       scope: {},
       templateUrl: 'richTextComponent/Video',
-      controller: ['$scope', '$attrs', 'explorationContextService',
-        function($scope, $attrs, explorationContextService) {
+      controller: ['$scope', '$attrs', 'explorationContextService', '$element',
+        'PAGE_CONTEXT',
+        function($scope, $attrs, explorationContextService, $element) {
           var start = oppiaHtmlEscaper.escapedJsonToObj($attrs.startWithValue);
           var end = oppiaHtmlEscaper.escapedJsonToObj($attrs.endWithValue);
 
-          $scope.autoplaySuffix = (oppiaHtmlEscaper.escapedJsonToObj(
-            $attrs.autoplayWithValue) ? '&autoplay=1' : '&autoplay=0');
           $scope.videoId = oppiaHtmlEscaper.escapedJsonToObj(
             $attrs.videoIdWithValue);
           $scope.timingParams = '&start=' + start + '&end=' + end;
+
+          // This code helps in checking whether video is for upcoming content.
+          // This is totally dependent of parent div of video container
+          // If in future that parent div changes then this code will be
+          // required to change accordingly.
+          var isVisible = true;
+          var videoVisibilty = angular.element($element).prop('offsetParent');
```

Comment:
I have checked it for all like Top, Height, Width there is no difference in all of them. Like top of hidden content will be same as Top of video in the first card. Width and Height for each of them is same too.